utils/azureUtilityManager.ts

```diff
@@ -0,0 +1,138 @@
+import { ContainerRegistryManagementClient } from 'azure-arm-containerregistry';
+import { ResourceManagementClient, SubscriptionModels } from 'azure-arm-resource';
+import { ResourceGroup } from "azure-arm-resource/lib/resource/models";
+import { ServiceClientCredentials } from 'ms-rest';
+import * as ContainerModels from '../node_modules/azure-arm-containerregistry/lib/models';
+import { AzureAccount } from '../typings/azure-account.api';
+import { AsyncPool } from '../utils/asyncpool';
+import { MAX_CONCURRENT_SUBSCRIPTON_REQUESTS } from './constants';
+
+/* Singleton for facilitating communication with Azure account services by providing extended shared
+  functionality and extension wide access to azureAccount. Tool for internal use.
+  Authors: Esteban Rey L, Jackson Stokes
+*/
+
+export class AzureUtilityManager {
+
+    //SETUP
+    private static _instance: AzureUtilityManager;
+    private azureAccount: AzureAccount;
+
+    private constructor() { }
+
+    public static getInstance(): AzureUtilityManager {
+        if (!AzureUtilityManager._instance) { // lazy initialization
+            AzureUtilityManager._instance = new AzureUtilityManager();
+        }
+        return AzureUtilityManager._instance;
+    }
+
+    //This function has to be called explicitly before using the singleton.
+    public setAccount(azureAccount: AzureAccount): void {
+        this.azureAccount = azureAccount;
+    }
+
+    //GETTERS
+    public getAccount(): AzureAccount {
+        if (this.azureAccount) { return this.azureAccount; }
+        throw new Error(('Azure account is not present, you may have forgotten to call setAccount'));
```

Comment:
Seems like you have three states:
1) Azure account extension available, setAccount() called
2) Azure account extension available, setAccount() not called
3) Azure account extension not available, setAccount() can't be called, getAccount() will throw if called, but no way for caller to know this